examples/kuka_iiwa_arm/kuka_simulation.cc

```diff
@@ -0,0 +1,228 @@
+                  world_plant->get_actuation_input_port(iiwa_model));
+
+  // Interpolating desired positions into desired velocities for controller.
+  builder.Connect(command_receiver->get_commanded_position_output_port(),
+                  desired_state_from_position->get_input_port());
+
+  // Connecting iiwa controller input ports
+  builder.Connect(world_plant->get_state_output_port(iiwa_model),
+                  iiwa_controller->get_input_port_estimated_state());
+  builder.Connect(desired_state_from_position->get_output_port(),
+                  iiwa_controller->get_input_port_desired_state());
+  builder.Connect(command_receiver->get_commanded_torque_output_port(),
+                  iiwa_controller->get_input_port_commanded_torque());
+
+  // Demux is for separating q and v from state output port.
+  builder.Connect(world_plant->get_state_output_port(iiwa_model),
+                  demux->get_input_port(0));
+
+  // Connecting outputs to iiwa state broadcaster
+  builder.Connect(demux->get_output_port(0),
+                  iiwa_status->get_position_measured_input_port());
+  builder.Connect(demux->get_output_port(1),
+                  iiwa_status->get_velocity_estimated_input_port());
+  builder.Connect(command_receiver->get_output_port(0),
+                  iiwa_status->get_position_commanded_input_port());
+  builder.Connect(iiwa_controller->get_output_port_control(),
+                  iiwa_status->get_torque_commanded_input_port());
+  builder.Connect(iiwa_controller->get_output_port_control(),
+                  iiwa_status->get_torque_measured_input_port());
+  builder.Connect(world_plant->get_generalized_contact_forces_output_port(iiwa_model),
+                  iiwa_status->get_torque_external_input_port());
+  builder.Connect(iiwa_status->get_output_port(),
+                  iiwa_status_publisher->get_input_port());
+
+  // Connecting world to scene graph components for visualization
+  builder.Connect(
+      world_plant->get_geometry_poses_output_port(),
+      scene_graph->get_source_pose_port(world_plant->get_source_id().value()));
+  builder.Connect(scene_graph->get_query_output_port(),
+                  world_plant->get_geometry_query_input_port());
+
+  drake::geometry::ConnectDrakeVisualizer(&builder, *scene_graph,
+      scene_graph->get_pose_bundle_output_port());
+
+  // Set the iiwa default joint configuration.
+  drake::VectorX<double> q0_iiwa(num_iiwa_positions);
+  q0_iiwa << 0, 0, 0, 0, 0, 0, 0;
+  const auto iiwa_joint_indices = world_plant->GetJointIndices(iiwa_model);
+  int q0_index = 0;
+  for (const auto joint_index : iiwa_joint_indices) {
+    drake::multibody::RevoluteJoint<double>* joint =
+        dynamic_cast<drake::multibody::RevoluteJoint<double>*>(
+            &world_plant->get_mutable_joint(joint_index));
+    // Note: iiwa_joint_indices includes the WeldJoint at the base.  Only set
+    // the RevoluteJoints.
+    if (joint) {
+      joint->set_default_angle(q0_iiwa[q0_index++]);
```

Comment:
Seems like a strange way to set initial condition. See other simulators (e.g. Cassie) for alternatives.